### ISSUE TITLE
is there any suggestion or plan to integrate selenide ?

### ISSUE THREAD

**Issue by bitlin** [MohabMohie/SHAFT_ENGINE#517]
selenide is a popular UI automation library in Java, is there any suggestion or plan to integrate selenide ?

**Comment by MohabMohie** [comment]
Hello @bitlin ,
We examined https://github.com/selenide/selenide before but we don't have any plans to integrate with it.
Our strategy is to integrate with base frameworks (Selenium rather than Selenide, RestAssured rather than Karate).
However we have been closely examining the use of DSL language that Selenide has in place, and @MhmdElGazzar has proposed and is the owner of implementing this approach in SHAFT.

This is the initial PR that @MhmdElGazzar has created and merged:
https://github.com/MohabMohie/SHAFT_ENGINE/pull/509

And this is the sample class that shows how the new syntax can be used:
https://github.com/MohabMohie/SHAFT_ENGINE/blob/master/src/test/java/dsl/testDsl.java

Let us know which features specifically caught your eye and we can share our plans with you in that regard.

**Comment by bitlin** [comment]
@MohabMohie   Thanks for your detail message. I really like this Automation Architecture, however, it 's more powerful and easy to use when using tools like selenide or karate and also FluentLenium framework. 

In fact, I am concerning the element locator enhancement right now. For example:

**as for selenide** , the elements locator based on their text
- ByText
- WithText
https://github.com/selenide/selenide/issues/1650

**as for fluentlenium, the additional filter to the Fluent Locator、page object pattern、 page injection pattern 

https://fluentlenium.com/docs/locators/#filters 
https://fluentlenium.com/docs/key_features/

** as for karate** 
Many functions are indeed more easy to use with very detailed document.

**as for  jdi-light** 
The test case writing pattern is very clear, maybe I mean the case structure 

https://github.com/jdi-templates/jdi-light-testng-template

**Another question?** 
If I just only add the above library or framework dependency to the SHAFT_ENGINE project, is there any features lost? for example, the allure report feature and so on.

More important, I'd like to hope this SHAFT_ENGINE project being one of the most popular automation platform.

**Comment by MohabMohie** [comment]
@bitlin sorry for dropping the ball on your comment, it's been a rough couple of weeks.

So, regarding "element locator enhancement" I haven't been considering this part, mainly because I was aware that Selenium 4 comes with Relative locators, and I didn't want to build wrappers and then have to maintain or destroy them completely for extended element identification. This is comparable to Appium's AppiumBy and MobileBy situation, and most of the custom locators that they were using that they had to remove for not being W3C compliant.

To answer your important question:
Nope, none of the existing features will be lost. But for example if you write an action using native selenium then you need to handle synchronization and logging for that action manually. Yet, regarding element identification, there are no wrappers in SHAFT around the native "By" object, so you are free to use any wrappers with ZERO impact on the stability and existing features.

In short, You can add any libraries you want to your code, and please share some code samples so that we can learn from them!

**Comment by hernandezmshawntae** [comment]
Please and Thank You.

**Comment by MohabMohie** [comment]
will close this issue due to lack of activity for the past week, but feel free to reopen it and keep the conversation going please :)
